### DISCUSSION TITLE
0.9.28.0 release

### DISCUSSION THREAD

**Discussion by kapilt** [cloud-custodian/cloud-custodian#8689] [Announcements]
# TLDR

New oracle cloud infrastructure provider, several additional resources to gcp, regular fixes and updates to extant providers.

# aws
 - aws - add delete action to directory and cloud-directory (#8610)
 - aws - add ses-receipt-rule resource and delete action (#8671)
 - aws - asg - fix propagate-tags for asgs with no tags (#8612)
 - aws - core - fix fetching resources by id for types with scalar server-side filters (#8614)
 - aws - ebs - encrypt-instance-volumes handle missing tags (#8683)
 - aws - efs-mount-target - support cloudtrail mode (#8631)
 - aws - event bus delete action (#8598)
 - aws - iam-user - add include-via option to policy filter for group inherited policies (#8372)
 - aws - security-group - used filter - handle ram vpc sharing eni when run in vpc owner (#8604)
 - aws - ses - add ses-email-identity resource type (#8616)
 - aws - ses - add set-delivery-options action (#8635)
 - aws - ses - identity has-statement filter (#8640)
 - aws - user-pool - include resource details from parent augment (#8684)
 - aws - vpc - metrics filter for vpce and tgw attachment (#8674)

# azure
 - azure - add recovery services vault resource (#8599)
 - azure - filter for the SQL Server TDE (#8652)
 - azure - monitor-log-profile resource (#8580)
 - azure - session - add _run_command timeout parameter (#8632)

# docs
 - docs - document gcp env vars explicitly along with noting workload federated identity support (#8606)
 - docs - update tencent cloud object storage example (#8600)

# gcp
 - gcp - add big table asset types metadata (#8615)
 - gcp - add datafusion resource (#8676)
 - gcp - add more bigtable resources (instance, cluster, table, backup) (#8519)
 - gcp - add support for impersonated credentials (#8571)
 - gcp - added notebook resource (#8680)
 - gcp - new resources app service and app service version (#8425)

# oci
 - oci - new provider (#8620)

# releng
 - releng - bump cryptography from 40.0.2 to 41.0.0 (#8627)
 - releng - bump cryptography from 40.0.2 to 41.0.0 in /tools/c7n_azure (#8626)
 - releng - bump cryptography from 40.0.2 to 41.0.0 in /tools/c7n_openstack (#8625)
 - releng - docker fix c7n-left image build via poetry install for dependency group changes (#8611)
 - releng - format c7n_left and c7n_mailer python code with black "make format" command (#8636)
 - releng - release prep for 0.9.28.0 (#8658)
 - releng - ruff whitespace config and github output (#8649)
 - releng - update issue template for new providers (#8655)

# tools
 - tools/c7n_mailer - refactoring and fix SendGrid duplicated emails (#8642)
 - tools/c7n_mailer - strip newlines from slack token (#8645)
 - tools/ops - fix mugc.py policy filtering  (#8670)

# schema changes
- [`aws.ses-email-identity`](https://cloudcustodian.io/docs/aws/resources/ses-email-identity.html) added
- [`aws.ses-receipt-rule-set`](https://cloudcustodian.io/docs/aws/resources/ses-receipt-rule-set.html) added
- [`azure.monitor-log-profile`](https://cloudcustodian.io/docs/azure/resources/monitor-log-profile.html) added
- [`azure.recovery-services`](https://cloudcustodian.io/docs/azure/resources/recovery-services.html) added
- [`gcp.app-engine-service`](https://cloudcustodian.io/docs/gcp/resources/app-engine-service.html) added
- [`gcp.app-engine-service-version`](https://cloudcustodian.io/docs/gcp/resources/app-engine-service-version.html) added
- [`gcp.bigtable-instance`](https://cloudcustodian.io/docs/gcp/resources/bigtable-instance.html) added
- [`gcp.bigtable-instance-cluster`](https://cloudcustodian.io/docs/gcp/resources/bigtable-instance-cluster.html) added
- [`gcp.bigtable-instance-cluster-backup`](https://cloudcustodian.io/docs/gcp/resources/bigtable-instance-cluster-backup.html) added
- [`gcp.bigtable-instance-table`](https://cloudcustodian.io/docs/gcp/resources/bigtable-instance-table.html) added
- [`gcp.datafusion-instance`](https://cloudcustodian.io/docs/gcp/resources/datafusion-instance.html) added
- [`gcp.notebook`](https://cloudcustodian.io/docs/gcp/resources/notebook.html) added
- [`oci.bucket`](https://cloudcustodian.io/docs/oci/resources/bucket.html) added
- [`oci.compartment`](https://cloudcustodian.io/docs/oci/resources/compartment.html) added
- [`oci.cross_connect`](https://cloudcustodian.io/docs/oci/resources/cross_connect.html) added
- [`oci.group`](https://cloudcustodian.io/docs/oci/resources/group.html) added
- [`oci.instance`](https://cloudcustodian.io/docs/oci/resources/instance.html) added
- [`oci.subnet`](https://cloudcustodian.io/docs/oci/resources/subnet.html) added
- [`oci.user`](https://cloudcustodian.io/docs/oci/resources/user.html) added
- [`oci.vcn`](https://cloudcustodian.io/docs/oci/resources/vcn.html) added
- [`oci.zone`](https://cloudcustodian.io/docs/oci/resources/zone.html) added
- `aws.artifact-repo`
  - added filters: [`json-diff`](https://cloudcustodian.io/docs/aws/resources/artifact-repo.html#aws-artifact-repo-filters-json-diff)
- `aws.cloud-directory`
  - added actions: [`delete`](https://cloudcustodian.io/docs/aws/resources/cloud-directory.html#aws-cloud-directory-actions-delete), [`disable`](https://cloudcustodian.io/docs/aws/resources/cloud-directory.html#aws-cloud-directory-actions-disable)
- `aws.directory`
  - added actions: [`delete`](https://cloudcustodian.io/docs/aws/resources/directory.html#aws-directory-actions-delete)
- `aws.ec2-spot-fleet-request`
  - added filters: [`json-diff`](https://cloudcustodian.io/docs/aws/resources/ec2-spot-fleet-request.html#aws-ec2-spot-fleet-request-filters-json-diff)
- `aws.event-bus`
  - added actions: [`delete`](https://cloudcustodian.io/docs/aws/resources/event-bus.html#aws-event-bus-actions-delete)
- `aws.firehose`
  - added filters: [`json-diff`](https://cloudcustodian.io/docs/aws/resources/firehose.html#aws-firehose-filters-json-diff)
- `aws.iam-certificate`
  - added filters: [`config-compliance`](https://cloudcustodian.io/docs/aws/resources/iam-certificate.html#aws-iam-certificate-filters-config-compliance), [`json-diff`](https://cloudcustodian.io/docs/aws/resources/iam-certificate.html#aws-iam-certificate-filters-json-diff)
- `aws.iam-saml-provider`
  - added filters: [`config-compliance`](https://cloudcustodian.io/docs/aws/resources/iam-saml-provider.html#aws-iam-saml-provider-filters-config-compliance), [`json-diff`](https://cloudcustodian.io/docs/aws/resources/iam-saml-provider.html#aws-iam-saml-provider-filters-json-diff)
- `aws.prefix-list`
  - added filters: [`config-compliance`](https://cloudcustodian.io/docs/aws/resources/prefix-list.html#aws-prefix-list-filters-config-compliance), [`json-diff`](https://cloudcustodian.io/docs/aws/resources/prefix-list.html#aws-prefix-list-filters-json-diff)
- `aws.ses-configuration-set`
  - added actions: [`set-delivery-options`](https://cloudcustodian.io/docs/aws/resources/ses-configuration-set.html#aws-ses-configuration-set-actions-set-delivery-options)
- `aws.transit-attachment`
  - added filters: [`metrics`](https://cloudcustodian.io/docs/aws/resources/transit-attachment.html#aws-transit-attachment-filters-metrics)
- `aws.vpc-endpoint`
  - added filters: [`metrics`](https://cloudcustodian.io/docs/aws/resources/vpc-endpoint.html#aws-vpc-endpoint-filters-metrics)
- `azure.sql-server`
  - added filters: [`transparent-data-encryption`](https://cloudcustodian.io/docs/azure/resources/sql-server.html#azure-sql-server-filters-transparent-data-encryption)
